control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1352,6 +1355,51 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile Azure cloud config with credentials: %w", err)
 		}
+	case hyperv1.KubevirtPlatform:
+		ownerRef := config.OwnerRefFrom(hcp)
+		sa := kubevirt.KubevirtCloudProviderServiceAccount(hcp.Namespace)
+		if _, err := r.CreateOrUpdate(ctx, r, sa, func() error {
+			return kubevirt.ReconcileKubevirtCloudProviderServiceAccount(sa, ownerRef)
+		}); err != nil {
+			return fmt.Errorf("failed to reconcile Kubevirt cloud provider service account: %w", err)
+		}
+		role := kubevirt.KubevirtCloudProviderRole(hcp.Namespace)
+		if _, err := r.CreateOrUpdate(ctx, r, role, func() error {
+			return kubevirt.ReconcileKubevirtCloudProviderRole(role, ownerRef)
+		}); err != nil {
+			return fmt.Errorf("failed to reconcile Kubevirt cloud provider role: %w", err)
+		}
+		roleBinding := kubevirt.KubevirtCloudProviderRoleBinding(hcp.Namespace)
+		if _, err := r.CreateOrUpdate(ctx, r, roleBinding, func() error {
+			return kubevirt.ReconcileKubevirtCloudProviderRoleBinding(roleBinding, ownerRef, sa, role)
+		}); err != nil {
+			return fmt.Errorf("failed to reconcile Kubevirt cloud provider rolebinding: %w", err)
+		}
+
+		// clusterRole := kubevirt.KubevirtCloudProviderClusterRole()
+		// if _, err := r.CreateOrUpdate(ctx, r, clusterRole, func() error {
+		// 	return kubevirt.ReconcileKubevirtCloudProviderClusterRole(clusterRole)
+		// }); err != nil {
+		// 	return fmt.Errorf("failed to reconcile Kubevirt cloud provider cluster role: %w", err)
+		// }
+		// clusterRoleBinding := kubevirt.KubevirtCloudProviderClusterRoleBinding()
+		// if _, err := r.CreateOrUpdate(ctx, r, clusterRoleBinding, func() error {
+		// 	return kubevirt.ReconcileKubevirtCloudProviderClusterRoleBinding(clusterRoleBinding, sa, clusterRole)
+		// }); err != nil {
+		// 	return fmt.Errorf("failed to reconcile Kubevirt cloud provider cluster rolebinding: %w", err)
+		// }
```

Comment:
This ClusterRole is needed in order to have permissions to get the Nodes (ccm need to read the zone and region from the nodes)
This code is in comment because I had some issues debugging this, I'm investigating it now

Reply:
> so the assumption is that the mgmt cluster is the same cluster that hosts the kubevirt instances?

Yes, this is the assumption here

> Using SA tokens is not really a possibility btw, as they will ne removed in 4.12

I wasn't aware to that, can you point me to the related enhancement please?

Reply:
> I wasn't aware to that, can you point me to the related enhancement please?

No enhancement afaik, just an announcement: https://groups.google.com/a/redhat.com/g/aos-devel/c/0g8ZLov65K4/m/DaGd8LBRAQAJ

Reply:
Thanks, I will change the code according to this change